util/metadata.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+# Copyright (c) 2018-2022 The MobileCoin Foundation
+#
+# Generate metadata for GitHub Actions workflows.
+#
+
+set -e
+
+export TMPDIR=".tmp"
+
+is_set()
+{
+  var_name="${1}"
+
+  if [ -z "${!var_name}" ]; then
+    echo "${var_name} is not set." >&2
+    exit 1
+  fi
+}
+
+normalize_ref_name()
+{
+    name="${1}"
+
+    echo "${name}" | sed -E 's/(feature|release)\///' | sed -e 's/[._/]/-/g'
+}
+
+# check for github reference variables.
+is_set GITHUB_REF_NAME
+is_set GITHUB_REF_TYPE
+is_set GITHUB_RUN_NUMBER
+is_set GITHUB_SHA
+is_set GITHUB_OUTPUT
+
+# Make sure prefix is less that 18 characters or k8s limits.
+namespace_prefix="mc"
+sha="sha-${GITHUB_SHA:0:8}"
+
+# Set branch name. Where we get this ref is different for branch, pull_request or delete event
+branch="${GITHUB_REF_NAME}"
+
+# Override branch name with head when event is a PR
+if [[ -n "${GITHUB_HEAD_REF}" ]]
+then
+    branch="${GITHUB_HEAD_REF}"
+fi
+
+# Override branch with delete ref if set.
+if [[ -n "${DELETE_REF_NAME}" ]]
+then
+    branch="${DELETE_REF_NAME}"
+fi
+
+case "${GITHUB_REF_TYPE}" in
+    tag)
+        # check for valid tag and set outputs
+        version="${GITHUB_REF_NAME}"
+        if [[ ! "${version}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+.* ]]
```

Comment:
Do we need to think about this pattern here and support the `v[0-9]+` pattern that twix uses.   I kinda wish we had just used the a bit more standard major.minor.patch, but there's a lot of tags now so :shrug:

Reply:
Good catch! I made the minor and patch parts optional